src/lib/modes/standard_modes.hpp

```diff
@@ -67,7 +67,33 @@ static inline StandardMode getStandardModeFromNavState(uint8_t nav_state)
 	case vehicle_status_s::NAVIGATION_STATE_AUTO_LAND: return StandardMode::LAND;
 
 	case vehicle_status_s::NAVIGATION_STATE_AUTO_TAKEOFF: return StandardMode::TAKEOFF;
-		// Note: all other standard modes do not directly map, or are vehicle-type specific
+
+	case vehicle_status_s::NAVIGATION_STATE_ALTCTL:
+		if (is_vtol || vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+		    || vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+			return StandardMode::ALTITUDE_HOLD;
+		}
+
+		break;
+
+	case vehicle_status_s::NAVIGATION_STATE_POSCTL:
+		if (!is_vtol && vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
```

Comment:
A VTOL will either be in cruise mode or position hold, depending on its flight state. I don't think that is captured in this.

Reply:
Yes I excluded VTOL here because setting these might require a transition. We can add that later on but it requires a bit more logic.